File: inv_java/src/main/java/inv_java/weixin/Weixin.java
package inv_java.weixin;

import java.util.HashMap;

import com.google.gson.JsonObject;

import inv_java.tools.Http;

public class Weixin {
	
	public static final String corpID = "wx49ef3962c2f3a2c0";
	public static final String secret = "l4mKdYE-C1VCROkcjmYoMv8dOQ3w1mF35L0dePiRXbL_SW5MMG7AN5tnURhAGvmb";

	public static void send(Message msg) {
		String access_token;
        
		{
			String gettokenUrl = String.format("https://qyapi.weixin.qq.com/cgi-bin/gettoken?corpid=%s&corpsecret=%s", 
					Weixin.corpID, Weixin.secret);
	        Http http = new Http(gettokenUrl);
	        HashMap<String, Object> map = http.getMap();
	        access_token = (String) map.get("access_token");
		}
        
        {
            JsonObject text = new JsonObject();
            text.addProperty("content", "api发发发!!! " + msg.toString());
            
            JsonObject post = new JsonObject();
            post.addProperty("touser", "wupei");
            post.addProperty("toparty", "");
            post.addProperty("totag", "");
            post.addProperty("msgtype", "text");
            post.addProperty("agentid", 0);
            post.add("text", text);
            post.addProperty("safe", 0);
            
            String msgUrl = String.format("https://qyapi.weixin.qq.com/cgi-bin/message/send?access_token=%s", access_token);
            Http http = new Http(msgUrl, post);
	        HashMap<String, Object> map = http.getMap();
        }
	}
	
}

File: inv_node/core/weixin.js
/**
 * Created by wupei on 16/9/26.
 */

var util = require('util');
var utils = require('./utils');
var co = require('co');
var http = require('./http');
var assert = require('assert');

var weixin = {};

var corpID = 'wx49ef3962c2f3a2c0';
var secret = 'l4mKdYE-C1VCROkcjmYoMv8dOQ3w1mF35L0dePiRXbL_SW5MMG7AN5tnURhAGvmb';

weixin.send = function (msg) {
    co(function* () {
        var txt = msg.message;

        // var gettokenUrl = util.format("https://qyapi.weixin.qq.com/cgi-bin/gettoken?corpid=%s&corpsecret=%s", corpID, secret);
        // var body = yield http.get(gettokenUrl);
        // var access_token = JSON.parse(body).access_token;
        // assert(access_token, "access_token empty, body = %s", body);
        var access_token = "abc";

        var form = JSON.stringify({
            text : { content : txt },
            touser : msg.touser,
            toparty : "",
            totag : "",
            msgtype : "text",
            agentid : 0,
            safe : 0,
        });
        var msgUrl = util.format("https://qyapi.weixin.qq.com/cgi-bin/message/send?access_token=%s", access_token);
        var body = yield http.post(msgUrl, form);
        var bodyJson = JSON.parse(body);
        if (bodyJson.errcode == 40014) {

        } else if (bodyJson.errcode == 0) {

        }
        console.log(body);

    }).catch(function (err) {
        utils.error(err);
    });
};

module.exports = weixin;